app/ide-desktop/lib/dashboard/src/providers/LocalStorageProvider.tsx

```diff
@@ -27,7 +27,14 @@ export interface LocalStorageProviderProps extends Readonly<React.PropsWithChild
 /** A React Provider that lets components get the shortcut registry. */
 export default function LocalStorageProvider(props: LocalStorageProviderProps) {
   const { children } = props
-  const localStorage = React.useMemo(() => new LocalStorage(), [])
+  const [, rerender] = React.useState<object>({})
+  const localStorage = React.useMemo(
+    () =>
+      new LocalStorage(() => {
+        rerender({})
```

Comment:
is there a reason we need to unconditionally rerender no matter which `LocalStorage` key is set?

Reply:
Yes, because we need to propagate the changes for components

Reply:
right, just wondering whether it's necessary for every key - for example most of the other states (expanded columns, keyboard shortcuts) already set state as appropriate and hence they don't need a full rerender of the entire page.

although if it's been tested and there's no noticeable latency or anything when changing the other LocalStorage entries then there's no problem